src/main/java/teammates/ui/controller/InstructorCourseDetailsPageData.java

```diff
@@ -20,9 +24,132 @@ public InstructorCourseDetailsPageData(AccountAttributes account) {
+            ElementTag viewButton = createButton("View", "btn btn-default btn-xs", null, this.getCourseStudentDetailsLink(student),
+                                                 Const.Tooltips.COURSE_STUDENT_DETAILS, "tooltip", null, isDisabled);
+            
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+            ElementTag editButton = createButton("Edit", "btn btn-default btn-xs", null, this.getCourseStudentEditLink(student),
+                                                 Const.Tooltips.COURSE_STUDENT_EDIT, "tooltip", null, isDisabled);
+            
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+            ElementTag sendInviteButton = createButton("Send Invite", "btn btn-default btn-xs", null, this.getCourseStudentRemindLink(student),
+                                                 Const.Tooltips.COURSE_STUDENT_REMIND, "tooltip", "return toggleSendRegistrationKey()", isDisabled);
+       
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+            onClick = "return toggleDeleteStudentConfirmation('" + sanitizeForJs(student.course) + "','" + sanitizeForJs(student.name) + "')";
+            ElementTag deleteButton = createButton("Delete", "btn btn-default btn-xs", null, this.getCourseStudentDeleteLink(student),
+                                                   Const.Tooltips.COURSE_STUDENT_DELETE, "tooltip", onClick, isDisabled);
+            
+            ElementTag allRecordsButton = createButton("All Records", "btn btn-default btn-xs", null, this.getStudentRecordsLink(student),
+                                                   Const.Tooltips.COURSE_STUDENT_RECORDS, "tooltip", null, false);
+
+            row.actions.add(viewButton);
+            row.actions.add(editButton);
+            if (!student.isRegistered()) {
+                row.actions.add(sendInviteButton);
+            }
+            row.actions.add(deleteButton);
+            row.actions.add(allRecordsButton);
+            
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(student.section, Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+            ElementTag addCommentButton = createButton("Add Comment", "btn btn-default btn-xs cursor-default", null, "javascript:;",
+                                                       Const.Tooltips.COURSE_STUDENT_COMMENT, "tooltip", null, isDisabled);
+            
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(student.section, Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+            ElementTag addCommentsButton = createButton("<span class=\"caret\"></span><span class=\"sr-only\">Add comments</span>", 
```

Comment:
In case where you have to use HTML code, it may be a good idea to store it in a `String` with a variable name that explains why the reason for the HTML code.
Or a comment.